src/platform/iotjs_module_pwm-linux-general.inl.h

```diff
@@ -46,51 +46,84 @@ void iotjs_pwm_initialize() {
   DDLOG("PWM initialize");
 }
 
-
-// Set PWM period.
-bool SetPwmPeriod(iotjs_pwm_reqdata_t* req_data) {
-  char path[PWM_PATH_BUFFER_SIZE] = { 0 };
-  if (snprintf(path, PWM_PATH_BUFFER_SIZE - 1, PWM_PIN_FORMAT,
-               PWM_DEFAULT_CHIP_NUMBER, req_data->pin) < 0) {
-    return false;
+// Generate device path for specified PWM device.
+// The path may include node suffix if passed ('enable', 'period', 'duty_cycle')
+// Pointer to a allocated string is returned, or null in case of error.
+static char* generateDeviceSubpath(iotjs_pwm_reqdata_t* req_data,
+                                   const char* fileName) {
+  char* devicePath = NULL;
+  int size;
+  if ((size = asprintf(&devicePath, PWM_PIN_FORMAT, PWM_DEFAULT_CHIP_NUMBER,
```

Comment:
If `_GNU_SOURCE` is not defined, this line will generate an error. And you can use `snprintf` and `iotjs_string` instead of `asprintf`.